src/feruchemy/java/leaf/cosmere/feruchemy/client/utils/FeruchemyChargeThread.java

```diff
@@ -73,83 +73,95 @@ public void run() {
+                            else
+                            {
+                                metalmindCharges.put(item.getMetalType(), chargeToAdd);
+                            }
                         }
                     }
                 }
-            }
 
-            // all curio metalminds are counted
-            CuriosApi.getCuriosHelper().getEquippedCurios(mc.player)
-                    .map(mapper ->
-                    {
-                        for (int i = 0; i < mapper.getSlots(); i++)
+                // all curio metalminds are counted
+                CuriosApi.getCuriosHelper().getEquippedCurios(mc.player)
+                        .map(mapper ->
                         {
-                            if (mapper.getStackInSlot(i).getItem() instanceof ChargeableMetalCurioItem item)
+                            for (int i = 0; i < mapper.getSlots(); i++)
                             {
-                                if (item.getItemCategory() == FeruchemyItemGroups.METALMINDS)
+                                if (mapper.getStackInSlot(i).getItem() instanceof ChargeableMetalCurioItem item)
                                 {
-                                    Double chargeToAdd = (double) item.getCharge(mapper.getStackInSlot(i));
-                                    if (metalmindCharges.containsKey(item.getMetalType()))
+                                    if (item.getItemCategory() == FeruchemyItemGroups.METALMINDS)
                                     {
-                                        metalmindCharges.put(item.getMetalType(), metalmindCharges.get(item.getMetalType()) + chargeToAdd);
-                                    }
-                                    else
-                                    {
-                                        metalmindCharges.put(item.getMetalType(), chargeToAdd);
+                                        Double chargeToAdd = (double) item.getCharge(mapper.getStackInSlot(i));
+                                        if (metalmindCharges.containsKey(item.getMetalType()))
+                                        {
+                                            metalmindCharges.put(item.getMetalType(), metalmindCharges.get(item.getMetalType()) + chargeToAdd);
+                                        }
+                                        else
+                                        {
+                                            metalmindCharges.put(item.getMetalType(), chargeToAdd);
+                                        }
                                     }
                                 }
                             }
-                        }
-                        return true;
-                    });
+                            return true;
+                        });
 
-            lock.lock();
-            try {
-                feruchemyChargeMap.clear();
-                feruchemyChargeMap.putAll(metalmindCharges);
+                lock.lock();
```

Comment:
We should change this from lock to if(lock.tryLock()), so that the thread doesn't throw an exception when the main thread is accessing the info. That way it just skips when the main thread is using it.

We may also want to just force the main thread to check on a certain tick number and the count thread to check on another. Being extra sure they aren't interfering with eachother and still able to do their own jobs.

Reply:
I had a thought earlier, I'm unsure how guaranteed it is for the tick to not change mid-operation, but I imagine it would help somewhat, especially when combined with try lock. 
More likely is the main thread ticking over and then trying to access and iterate the list while the thread is still building it

Reply:
So that's a go on the odd-even tick split? Or just the tryLock()?

Reply:
Let's do both, so at least if the main thread gets to it first, the charge thread doesn't needlessly try modify.